src/HttpServer.ts

```diff
@@ -65,31 +65,30 @@ export class HttpServer {
     }
 
     // TODO: use this when changing to jsonrpc
-    async rootHandler(req: any, res: any): Promise<void> {
+    async rootHandler(req: Request, res: Response): Promise<void> {
         let status;
+        let id = -1;
         try {
-            let res;
-            // @ts-ignore
-            const func = this.backend[req.body.method];
-            if (func != null) {
-                res = (await func.apply(this.backend, [req.body.params])) ?? {
-                    code: 200
-                };
-            } else {
-                // @ts-ignore
-                // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-                throw Error(
-                    `Implementation of method ${req.body.params} not found on backend!`
-                );
+            if (
+                !Object.prototype.hasOwnProperty.call(req.body, 'id') ||
+                !Object.prototype.hasOwnProperty.call(req.body, 'method')
+            ) {
+                throw Error('Missing properties');
             }
-            status = jsonrpc.success(req.body.id, res);
+            id = req.body.id;
+            const result = (await this.processRootHandler(
+                req.body.method,
+                req.body.params
+            )) ?? { code: 200 };
```

Comment:
`{ code: 200 }` does not seem to be tested

Reply:
updated